ui/src/timeMachine/actions/queryBuilder.ts

```diff
@@ -148,9 +155,14 @@ export const selectAggregateWindow = (period: string) => (
 }
 
 export const loadBuckets = () => async (
-  dispatch: Dispatch<Action | ReturnType<typeof selectBucket>>,
+  dispatch: Dispatch<Action | ReturnType<typeof selectBucket> | BucketAction>,
   getState: GetState
 ) => {
+  if (
+    getStatus(getState(), ResourceType.Buckets) === RemoteDataState.NotStarted
```

Comment:
so this will work, and it's used in places in our app, so it's not a criticism of you using this pattern, just a heads up: this is a bit of an inference - we're inferring that the state is complete based on the remote data state. that's fine when the state is `Not Started`, but all that `Loaded` tells us is that he data has been loaded. It doesn't tell us what data is loaded, and whether it's been reloaded, whether the state is good, etc. i have a feeling this is going to become relevant soon as we start thinking about when to run and re-run queries and requests. no action item here but to think about how we know if we need to re-fetch data, in general (not just buckets)